internal/manifests/builder.go

```diff
@@ -39,3 +41,10 @@ func Factory[T client.Object](f ManifestFactory[T]) K8sManifestFactory {
 		return f(cfg, logger, otelcol)
 	}
 }
+
+// ShouldCreateObject ensures that we only create an object IFF it isn't nil
+// and it's concrete type isn't nil either. This works around the Go type system
+// by using reflection to verify its concrete type isn't nil.
+func ShouldCreateObject(obj client.Object) bool {
```

Comment:
While the name `ShouldCreateObject` makes sense everywhere in the caller context used here, individually I feel it is a bit misleading.

What about `IsNil` or `ObjectIsNil` (alternatively, the negative form).